hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/meta/CkpMetadata.java

```diff
@@ -132,6 +132,20 @@ private void clean(String newInstant) {
     }
   }
 
+  /**
+   * start a checkpoint commit message.
+   *
+   * @param instant The start commit instant
+   */
+  public void startCommitInstant(String instant) {
```

Comment:
The reason we do not support this is because we have no idea how many tasks are running for partial mode, and we have no idea when to commit the metadata data.

One solution is we commit the metadata in commit sink, but that is a different mechanism and we need to rethink about it.

Reply:
This method does not affect the metadata submission at the end of the commit. This is just a status to record the start of commit.

Reply:
I think the reason why the task is blocked is that it needs to be done when committing instant. However, it is not accurate to use inflight to represent running instant and committing instant instead of recording a state in commit.